openfisca_core/entities/typing.py

```diff
@@ -1,9 +1,18 @@
+from abc import abstractmethod
 from typing import Protocol
 
 
-class Entity(Protocol):
-    """Entity protocol.
+class HasKey(Protocol):
```

Comment:
I feel that the main value brought by the  `Entity` protocol is signalling the type by its name. I am not convinced by the value of making this generic. In my view, it makes the entities module less readable by adding a more generic name, and the protocol is so trivial it has little intrinsic value.

I thus suggest to keep the `Entity` name at this stage and rename it to a more generic name only if it gets extracted from the `entities` module.

Reply:
Makes sense. And you're right, the main use here is to "duck-reference" classes between them. That improves static analysis and avoids run time problems (cyclic imports, inheritance, etc).